sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1024,7 +1065,8 @@ std::unique_ptr<CurlNetworkConnection> CurlConnectionPool::GetCurlConnection(
     std::lock_guard<std::mutex> lock(CurlConnectionPool::ConnectionPoolMutex);
 
     // get a ref to the pool from the map of pools
-    auto hostPoolIndex = CurlConnectionPool::ConnectionPoolIndex.find(host);
+    auto hostPoolIndex
+        = CurlConnectionPool::ConnectionPoolIndex.find(host + connectionPropertiesKey);
```

Comment:
Can we change `connectionPropertiesKey` to actually be equivalent to `host + connectionPropertiesKey` and not just base it on the options solely?

Reply:
Implementation detail.. don't worry about it.

Reply:
Factors such as code maintenance, ease of understanding, and principles such as single responsibility still matters as part of code review, even if it is just an implementation detail :)

Reply:
Well, my general feeling is that it is not such a deal for our team.
I remember I mentioned the `single responsibility` for the winHttp transport adapter when I saw the handle manager as a tight coupling to anyone using it, and you mentioned it's just implementation and that's how you preferred. XD

Anyway, I have done this change :)

Reply:
> when I saw the handle manager as a tight coupling to anyone using it

I had added that based on your recommendation :)